core/src/main/java/com/linecorp/armeria/internal/common/DefaultRequestTarget.java

```diff
@@ -203,6 +203,8 @@ public static RequestTarget createWithoutValidation(
     private final String fragment;
     private boolean cached;
 
+    private static boolean isPreservedPercentEncoding;
```

Comment:
In general, I don't think we want the variable to be static since we want this option to be different for each client/serverConfig implementation